src/web/app/pages-instructor/instructor-session-individual-extension-page/instructor-session-individual-extension-page.component.ts

```diff
@@ -0,0 +1,528 @@
+    modalRef.componentInstance.selectedStudents = selectedStudents;
+    modalRef.componentInstance.selectedInstructors = selectedInstructors;
+    modalRef.componentInstance.extensionTimestamp = this.feedbackSessionEndingTimestamp;
+    modalRef.componentInstance.feedbackSessionTimeZone = this.feedbackSessionTimeZone;
+
+    modalRef.componentInstance.onConfirmExtensionCallBack.subscribe((isNotifyDeadlines: boolean) => {
+      this.handleCreateAndDeleteDeadlines(selectedStudents, selectedInstructors, DeadlineHandlerType.DELETE,
+        isNotifyDeadlines);
+      modalRef.componentInstance.isSubmitting = false;
+      modalRef.close();
+    });
+  }
+
+  private handleCreateAndDeleteDeadlines(
+    selectedStudents: StudentExtensionTableColumnModel[],
+    selectedInstructors: InstructorExtensionTableColumnModel[],
+    updateDeadlinesType: DeadlineHandlerType,
+    isNotifyDeadlines: boolean,
+    extensionTimestamp?: number,
+  ): void {
+    const request: FeedbackSessionUpdateRequest = {
+      studentDeadlines: this.getUpdatedDeadlines(selectedStudents, this.studentDeadlines,
+        updateDeadlinesType, extensionTimestamp),
+      instructorDeadlines: this.getUpdatedDeadlines(selectedInstructors, this.instructorDeadlines,
+        updateDeadlinesType, extensionTimestamp),
+      ...this.feedbackSessionDetails,
+    };
+
+    this.isSubmittingDeadlines = true;
+    this.feedbackSessionsService
+      .updateFeedbackSession(this.courseId, this.feedbackSessionName, request, isNotifyDeadlines)
+      .pipe(finalize(() => { this.isSubmittingDeadlines = false; }))
+      .subscribe(() => {
+          this.loadFeedbackSessionAndIndividuals();
+          this.showSucessToast(updateDeadlinesType, selectedStudents, selectedInstructors);
+        },
+        (resp: ErrorMessageOutput) => {
+          this.statusMessageService.showErrorToast(resp.error.message);
+        },
+      );
+  }
+
+  private showSucessToast(updateDeadlinesType: DeadlineHandlerType,
+    selectedStudents: StudentExtensionTableColumnModel[],
+    selectedInstructors: InstructorExtensionTableColumnModel[],
+  ): void {
+    const updateAction = updateDeadlinesType === DeadlineHandlerType.CREATE ? 'created' : 'deleted';
+    this.statusMessageService.showSuccessToast(
+      `Successfully ${updateAction} extension(s) for ${selectedStudents.length} student(s) and`
+    + ` ${selectedInstructors.length} instructor(s)!`);
+  }
+
+  private getUpdatedDeadlines(
+    selectedIndividuals: StudentExtensionTableColumnModel[] | InstructorExtensionTableColumnModel[],
+    deadlinesToCopyFrom: Record<string, number>,
+    updateDeadlinesType: DeadlineHandlerType,
+    extensionTimestamp?: number,
+  ): Record<string, number> {
+    const record: Record<string, number> = { ...deadlinesToCopyFrom };
+
+    if (updateDeadlinesType === DeadlineHandlerType.CREATE) {
+      selectedIndividuals.forEach((x) => {
+        record[x.email] = extensionTimestamp!;
```

Comment:
In the spirit of defensive programming, what would happen if `extensionTimestamp` is `null`?

Reply:
Yeah, it's hard to enforce or prevent it to be null after merging them. Would it be better then to split the function back into separate functions?

Reply:
Hmm that's a lot more code... I suppose this is fine as the wrong use would not make sense from the look of it (create deadline without timestamp)